### PR TITLE
Add tests for molecules linkcard

### LINKED ISSUE #36: Backfill tests for Molecules/LinkCard

### PR DESCRIPTION
**Pull request by SaidShah**
…t value for the to field in link card

- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)
- [x] New component and/ or properties? Make sure to update storybook

Closes #36

### CODE REVIEW

File: src/molecules/link-card/link-card.test.tsx [hunk 1]
```diff
@@ -1,3 +1,59 @@
+import React from "react";
+import faker from "faker";
+import { LinkCard } from "./link-card";
+import { TestUtils } from "../../tests/test-utils";
+import { LinkCardTypes } from "../constants/link-card-types";
+
 describe("LinkCard", () => {
-    test.skip("TODO - https://github.com/AndcultureCode/AndcultureCode.JavaScript.React.Components/issues/36", () => {});
+    test("When default props, renders LinkCard", () => {
```
Comment (brandongregoryscott) [inline src/molecules/link-card/link-card.test.tsx:8]: Nitpick - for consistency, we don't usually capitalize test names in JS/TS (references to components/proper nouns are fine)

File: src/molecules/link-card/link-card.test.tsx [hunk 2]
```diff
@@ -1,3 +1,59 @@
...
+        // Act
+        const { getByText } = TestUtils.renderWithRouter(
+            <LinkCard label={label}>{expected}</LinkCard>
+        );
+
+        //Assert
+        expect(getByText(expected)).not.toBeNull();
+    });
+
+    test("When type prop is button, renders LinkCard with button", () => {
```
Comment (brandongregoryscott) [inline src/molecules/link-card/link-card.test.tsx:22]: Could this be string interpolated with `LinkCardTypes.Button`?

File: src/molecules/link-card/link-card.test.tsx [hunk 3]
```diff
@@ -1,3 +1,59 @@
...
+            <LinkCard label={label} type={type}>
+                {expected}
+            </LinkCard>
+        );
+        const result = container.getElementsByTagName(LinkCardTypes.Button)[0];
+
+        // Assert
+        expect(result).not.toBeNull();
+        expect(result.classList).toContain("c-button");
+        expect(container.innerHTML).toContain("button");
```
Comment (brandongregoryscott) [inline src/molecules/link-card/link-card.test.tsx:39]: What is this testing, exactly?

File: src/molecules/link-card/link-card.test.tsx [hunk 4]
```diff
@@ -1,3 +1,59 @@
...
+        const { container } = TestUtils.renderWithRouter(
+            <LinkCard label={label} type={type}>
+                {expected}
+            </LinkCard>
+        );
+        const result = container.getElementsByTagName(LinkCardTypes.Button)[0];
+
+        // Assert
+        expect(result).not.toBeNull();
+        expect(result.classList).toContain("c-button");
```
Comment (brandongregoryscott) [inline src/molecules/link-card/link-card.test.tsx:38]: Could we export a constant for the button class name and reference it here? I believe we've done that somewhere else to cut down on hard-coded strings for styles across the library (`list-box.tsx` as an example)

File: src/tests/test-utils.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ * Returns a React component wrapped in a MemoryRouter. Used for Components that
+ * must be rendered inside a Route Component.
+ *
+ * @param component
+ * @returns {ReactComponent}
+ */
+
+const _renderWithRouter = (ui, { route = "/" } = {}) => {
```
Comment (brandongregoryscott) [inline src/tests/test-utils.ts:16]: What type is `ui` here? And the second param?

Otherwise, I love the idea 👍

**Review by brandongregoryscott — COMMENTED** [review]
Thanks for back-filling these tests @SaidShah! Some minor cleanup & questions I'd like answered before approving.

File: src/molecules/link-card/link-card.test.tsx (hunk 3 above)
Comment (SaidShah) [reply src/molecules/link-card/link-card.test.tsx:39]: This is testing that the LinkCard is rendered with a button and not an a tag

Comment (brandongregoryscott) [reply src/molecules/link-card/link-card.test.tsx:38]: I'm not too familiar with `react-testing-library`, but if we're retrieving the DOM element with `getElementsByTagName(LinkCardTypes.Button)` (where `LinkCardTypes.Button` is `Button`), and we know it's not `null` or `undefined`, shouldn't that already be covered?

Comment (SaidShah) [reply src/molecules/link-card/link-card.test.tsx]: Thanks for jumping on the call with me to discuss this. 👍

File: src/tests/test-utils.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * Returns a React component wrapped in a MemoryRouter. Used for Components that
+ * must be rendered inside a Route Component.
+ *
+ * @param component
+ * @returns {ReactComponent}
+ */
+
+const _renderWithRouter = (ui: ReactElement, route: string = "/") => {
```
Comment (brandongregoryscott) [inline src/tests/test-utils.ts:17]: @SaidShah Sorry for the back & forth - thanks for adding these types. Can you update the JSDoc comment to match? (Whether that's renaming the param or updating the comment - `ui` to me is a bit vague, `component` seems more appropriate)

**Review by brandongregoryscott — CHANGES_REQUESTED** [review]

**Review by brandongregoryscott — APPROVED** [review]

**Pull request merged**